src/todolistediting.js

```diff
@@ -73,19 +73,40 @@ export default class TodoListEditing extends Plugin {
 			modelViewInsertion( model, listItem => this._handleCheckmarkChange( listItem ) ),
 			{ priority: 'high' }
 		);
-		editing.downcastDispatcher.on( 'insert:$text', modelViewTextInsertion, { priority: 'high' } );
 		data.downcastDispatcher.on( 'insert:listItem', dataModelViewInsertion( model ), { priority: 'high' } );
 		data.downcastDispatcher.on( 'insert:$text', dataModelViewTextInsertion, { priority: 'high' } );
 
 		editing.downcastDispatcher.on(
 			'attribute:listType:listItem',
-			modelViewChangeType( listItem => this._handleCheckmarkChange( listItem ) )
+			modelViewChangeType( listItem => this._handleCheckmarkChange( listItem ), editing.view )
 		);
 		editing.downcastDispatcher.on(
 			'attribute:todoListChecked:listItem',
 			modelViewChangeChecked( listItem => this._handleCheckmarkChange( listItem ) )
 		);
 
+		// Fix view position on 0 offset.
+		editing.mapper.on( 'modelToViewPosition', ( evt, data ) => {
+			const view = editing.view;
+			const modelPosition = data.modelPosition;
+
+			if ( modelPosition.parent.is( 'listItem' ) && modelPosition.parent.getAttribute( 'listType' ) == 'todo' ) {
+				const viewLi = editing.mapper.toViewElement( data.modelPosition.parent );
+
+				if ( modelPosition.offset === 0 ) {
+					const label = findLabel( viewLi, view );
+
+					if ( label ) {
+						if ( label.nextSibling ) {
+							data.viewPosition = view.createPositionAt( label.nextSibling, 0 );
```

Comment:
You should ensure that nextSibling is not an UIElement. Selection converter will try to break it (that's why one unit test is red).

UIElement is used to represent a selection of collaborative users, so when two users put selection at the beginning of the `totoList` item it's gonna 💥 💥 💥

Reply:
Thanks for the clarification :) I'm still not used to the markers :/